lib/cloudant.js

```diff
@@ -1108,14 +1119,17 @@ function buildFilterArr(k, props) {
  * @param {Object} include Include filter
  * @param {Function} [cb] The cb function
  */
-function findRecursive(mo, query, docs, include, cb) {
+Cloudant.prototype._findRecursive = function(mo, query, docs, include, cb) {
+  var self = this;
   mo.db.find(query, function(err, rst) {
     debug('Cloudant.prototype.all (findRecursive) results: %j %j', err, rst);
     if (err) return cb(err);
 
     // only sort numeric id if the id type is of Number
-    if (mo.mo.properties.id.type.name === 'Number' && query.sort)
-      sortNumericId(rst.docs, query.sort);
+    var idName = self.getIdName(mo.mo.model.modelName);
+    if (!!idName && mo.mo.properties[idName].type.name === 'Number' &&
```

Comment:
@b-admike The purpose of `!!` is to convert undefined/null/string/....to a boolean value, so I think it's fine to keep it this way?